planner/core/preprocess_test.go

```diff
@@ -228,11 +231,14 @@ func (s *testValidatorSuite) TestValidator(c *C) {
 	}()
 	se, err := session.CreateSession4Test(store)
 	c.Assert(err, IsNil)
-	_, err = se.Execute(context.Background(), "use test")
-	c.Assert(err, IsNil)
 	ctx := se.(sessionctx.Context)
 	is := infoschema.MockInfoSchema([]*model.TableInfo{core.MockSignedTable()})
-	for _, tt := range tests {
+	for k, tt := range tests {
+		// the first test should be executed without selecting a database
+		if k == 1 {
+			_, err = se.Execute(context.Background(), "use test")
+			c.Assert(err, IsNil)
+		}
```

Comment:
I prefer to move the new test to a separate test case, instead of hacking it here.
Besides, 3 cases need to be tested:
1. Add constraint after `use`.
2. Add constraint without `use`.
3. Add constraint with a specified schema before the table name.

Reply:
It is separated. Is there any problem with the new tests?